core/arch/arm/plat-sam/freq.c

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2022 Microchip
+ */
+
+#include <drivers/clk.h>
+#include <drivers/clk_dt.h>
+#include <kernel/boot.h>
+#include <kernel/panic.h>
+#include <libfdt.h>
+
+static unsigned long freq;
+
+static unsigned long get_freq_from_dt(void)
```

Comment:
I'd prefer doing this once at boot so we can forget about the embedded DT once boot is done.

Reply:
Since I'm using the clock framework, it is not ready at boot time but just after the clock framework is probed. Doing so allows to defer the clock retrieval.

Reply:
The point is that I'd like to avoid depending on an embedded DT once boot is done. Let's say that the embedded DT is replaced by a DT which is only available until we do the first exit to normal world.

Reply:
Hum ok, I understand. So indeed, I will try to add an early initicall to retrieve the frequency from the DT with the clock framework once and thus setup the plat_freq variable at init only.